zerver/tests/test_muting_users.py

```diff
@@ -220,3 +220,33 @@ def test_new_messages_from_muted_user_marked_as_read(self) -> None:
         pm_to_hamlet = self.send_personal_message(cordelia, hamlet, "Spam in PM")
         private_usermessage = self.get_user_message(hamlet, pm_to_hamlet)
         self.assertTrue(private_usermessage.flags.read)
+
+    def test_existing_messages_from_muted_user_marked_as_read(self) -> None:
+        hamlet = self.example_user("hamlet")
+        self.login_user(hamlet)
+        cordelia = self.example_user("cordelia")
+        iago = self.example_user("iago")
+
+        self.make_stream("general")
+        self.subscribe(hamlet, "general")
+        self.subscribe(cordelia, "general")
+        self.subscribe(iago, "general")
+
+        # Have Cordelia send messages to Hamlet, then have Hamlet mute Cordelia.
+        # Then test that the messages sent earlier got marked as read.
+        stream_message = self.send_stream_message(cordelia, "general", "Spam in stream")
+        huddle_message = self.send_huddle_message(cordelia, [hamlet, iago], "Spam in huddle")
+        pm_to_hamlet = self.send_personal_message(cordelia, hamlet, "Spam in PM")
```

Comment:
This test should verify the messages are unread before the operation.  Possibly makes sense to factor out a function we can call twice, maybe `assert_usermessages_read_flag(messages, True|False)` to avoid messy duplication.